### PR TITLE
Remove ClusterResourceSet dependency in e2e up

### LINKED ISSUE #55: make e2e-up fails to create capi-dns namespace when using cluster-api v0.3.12
## Short Description

Currently, when I run `make e2e-up` I see an error indicating that the `capi-dns` namespace does not exist. This same error then causes `make test-connectivity-api` to fail.

See [this check run](https://github.com/vmware-tanzu/cross-cluster-connectivity/runs/1644788818#step:9:168) as an example.

However, [the same check runs on the same commit](https://github.com/vmware-tanzu/cross-cluster-connectivity/runs/1525877867) passed on December 9th

I'm guessing some runtime in our tests that isn't pinned to a particular version has changed in a breaking way but I haven't tracked it down yet.

## Steps To Reproduce

```
make build-images
make e2e-up
```

Output snippet [toward the end](https://github.com/vmware-tanzu/cross-cluster-connectivity/runs/1644788818#step:9:168):

```
Error from server (NotFound): namespaces "capi-dns" not found
configmap/coredns patched (no change)
Error from server (NotFound): namespaces "capi-dns" not found
configmap/coredns patched (no change)
```

## Expected Behavior

`make e2e-up` succeeds without that error printing, and then `make test-connectivity-api-dns` passes.

## Environment:
 - Cross-cluster connectivity version: 750c8d965b433229761cef5418daef339870745e
 - Kubernetes version: (use `kubectl version`):
 - Cloud provider or hardware configuration: 

```
$ kubectl --context kind-management version
Client Version: version.Info{Major:"1", Minor:"16", GitVersion:"v1.16.4", GitCommit:"224be7bdce5a9dd0c2fd0d46b83865648e2fe0ba", GitTreeState:"clean", BuildDate:"2019-12-11T12:47:40Z", GoVersion:"go1.12.12", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"19", GitVersion:"v1.19.1", GitCommit:"206bcadf021e76c27513500ca24182692aabd17e", GitTreeState:"clean", BuildDate:"2020-09-14T07:30:52Z", GoVersion:"go1.15", Compiler:"gc", Platform:"linux/amd64"}

$ kubectl --kubeconfig cluster-a.kubeconfig version
Client Version: version.Info{Major:"1", Minor:"16", GitVersion:"v1.16.4", GitCommit:"224be7bdce5a9dd0c2fd0d46b83865648e2fe0ba", GitTreeState:"clean", BuildDate:"2019-12-11T12:47:40Z", GoVersion:"go1.12.12", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"17", GitVersion:"v1.17.2", GitCommit:"59603c6e503c87169aea6106f57b9f242f64df89", GitTreeState:"clean", BuildDate:"2020-02-07T01:05:17Z", GoVersion:"go1.13.5", Compiler:"gc", Platform:"linux/amd64"}

$ kubectl --kubeconfig cluster-b.kubeconfig version
Client Version: version.Info{Major:"1", Minor:"16", GitVersion:"v1.16.4", GitCommit:"224be7bdce5a9dd0c2fd0d46b83865648e2fe0ba", GitTreeState:"clean", BuildDate:"2019-12-11T12:47:40Z", GoVersion:"go1.12.12", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"17", GitVersion:"v1.17.2", GitCommit:"59603c6e503c87169aea6106f57b9f242f64df89", GitTreeState:"clean", BuildDate:"2020-02-07T01:05:17Z", GoVersion:"go1.13.5", Compiler:"gc", Platform:"linux/amd64"}

$ clusterctl version
clusterctl version: &version.Info{Major:"0", Minor:"3", GitVersion:"v0.3.10", GitCommit:"af6630920560ca0e12179897b96d6ea8bd830b63", GitTreeState:"clean", BuildDate:"2020-10-01T14:29:50Z", GoVersion:"go1.13.15", Compiler:"gc", Platform:"linux/amd64"}

$ kind version
kind v0.9.0 go1.15.2 linux/amd64
```

### PR DESCRIPTION
**Pull request by mcwumbly**
- Our walk-through docs do not use ClusterResourceSet, and there is no
major advantage to using them in our e2e scripts.
- ClusterResourceSet is a bleeding edge feature of Cluster API
- This resolves #55 and allows us to use the latest version of Cluster
API

Fixes #55

### CODE REVIEW

File: hack/e2e.sh [hunk 1]
```diff
@@ -144,37 +142,6 @@ function setup_management_cluster() {
 
   clusterctl init \
     --kubeconfig-context "kind-${KIND_MANAGEMENT_CLUSTER}" \
```
Comment (tylerschultz) [inline hack/e2e.sh:144]: Is the trailing slash still needed?

**Comment by tylerschultz** [comment]
This looks great Dave, it really simplifies the script. One thing that wasn't obvious about the way it was written before was that it was more parallelized. @christianang created a PR recently to add this parallelization. The cluster creation was kicked off for both clusters, then the script waited for the clusters to come up. If I understand correctly, this change will go back to a more serial approach. On the other hand, there's less happening in this script, so it may be faster anyways.

**Comment by tylerschultz** [comment]
I'm seeing now that you and @christianang appear to have already sorted out the parallelization in different PRs.

**Review by tylerschultz — APPROVED** [review]

File: hack/e2e.sh (hunk 1 above)
Comment (mcwumbly) [reply hack/e2e.sh]: good catch, removed.

**Comment by mcwumbly** [comment]
Regarding the parallelization that was added recently and will be added back in #73, I haven't seen that it impacts the speed of e2e-up much one way or another in practice. 

Something to consider if in the future we want to either a) remove it to keep things simpler and/or b) dig deeper into where time is actually going and try to speed things up.

**Pull request merged**